### PR TITLE
crate rhymes with u8, now compiles in 2018 edition

### PR DESCRIPTION
**Pull request by s-e**

### CODE REVIEW

**Comment by jyn514** [comment]
`weird-exprs` does strange things like this intentionally; just because we _can_ doesn't necessarily mean it's a useful change. Can you explain why this is better than previous behavior?

File: src/test/ui/weird-exprs.rs [hunk 1]
```diff
@@ -99,7 +99,7 @@ fn u8(u8: u8) {
 
             u8!(u8);
             let &u8: &u8 = u8::u8(&8u8);
-            ::u8(0u8);
+            crate::u8(0u8);
```
Comment (memoryruins) [inline src/test/ui/weird-exprs.rs:102]: To make the rhyming intentions clear:
```rust
use crate as cr8;

fn u8(u8: u8) -> u8 {
    /* ... */
    crate::cr8::u8(0u8);
    u8
}
```

**Comment by s-e** [comment]
@jyn514 apologies this was linked in a discord as a curiosity; and out of curiosity I tried to compile it with nightly and out of test context, not realising it might not be tested under 2018 edition. @memoryruins has since informed me that the ui crate isn't necessarily tested in this context and has its own special test rules.

short version: I made this change just to get it to compile and submitted it as a PR without realising the bigger picture, feel free to close this.

**Comment by jyn514** [comment]
Gotcha, thanks for trying to contribute :) I'll close this then but if you're interested in working in something else I'm happy to help!

**Pull request closed without merging**